### PR TITLE
ipc_service: Fix API documentation

### PR DESCRIPTION
**Pull request by carlocaione**
clarify that `cfg` must be kept alive when the endpoint is in use and fix some retvalue in the API documentation.

### CODE REVIEW

File: include/ipc/ipc_service.h [hunk 1]
```diff
@@ -142,6 +142,8 @@ int ipc_service_open_instance(const struct device *instance);
  *  @param ept Endpoint object.
  *  @param cfg Endpoint configuration.
  *
+ *  @note Keep the variable pointed by @em cfg alive when endpoint is in use.
```
Comment (gmarull) [inline include/ipc/ipc_service.h:145]:
```suggestion
 *  @note Keep the variable pointed by @p cfg alive when endpoint is in use.
```

File: include/ipc/ipc_service.h [hunk 2]
```diff
@@ -165,7 +167,7 @@ int ipc_service_register_endpoint(const struct device *instance,
  *  @retval -EBADMSG when the message is invalid.
  *  @retval -EBUSY when the instance is busy.
  *
- *  @retval 0 on success.
+ *  @retval number of bytes it has sent.
```
Comment (gmarull) [inline include/ipc/ipc_service.h:170]: nit: `@retval` syntax is `@retval VALUE DESCRIPTION`, so maybe:

```suggestion
 *  @retval bytes Number of bytes it has sent.
```

Comment (carlocaione) [reply include/ipc/ipc_service.h:170]: the least number of lines I change the most number of errors I make :D

**Review by gmarull — APPROVED** [review]

**Comment by carlocaione** [comment]
Can we ack / merge this?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**